templates/sshd_config.erb

```diff
@@ -196,6 +196,10 @@ MaxSessions <%= @sshd_config_maxsessions_integer %>
 #MaxSessions 10
 <% end -%>
 
+<% if @sshd_config_authenticationmethods != [] -%>
+AuthenticationMethods  <%= @sshd_config_authenticationmethods.join(' ') -%>
```

Comment:
The join character should be `,` instead of a space.
Quoting the man page of sshd(5)

>      AuthenticationMethods
> 	     Specifies the authentication methods that must be successfully
> 	     completed for a user to be	granted	access.	 This option must be
> 	     followed by one or	more comma-separated lists of authentication
> 	     method names.  Successful authentication requires completion of
> 	     every method in at	least one of these lists.
> 
> 	     For example, an argument of ``publickey,password
> 	     publickey,keyboard-interactive'' would require the	user to	com-
> 	     plete public key authentication, followed by either password or
> 	     keyboard interactive authentication.  Only	methods	that are next
> 	     in	one or more lists are offered at each stage, so	for this exam-
> 	     ple, it would not be possible to attempt password or keyboard-
> 	     interactive authentication	before public key.
> 
> 	     For keyboard interactive authentication it	is also	possible to
> 	     restrict authentication to	a specific device by appending a colon
> 	     followed by the device identifier ``bsdauth'', ``pam'', or
> 	     ``skey'', depending on the	server configuration.  For example,
> 	     ``keyboard-interactive:bsdauth'' would restrict keyboard interac-
> 	     tive authentication to the	``bsdauth'' device.
> 
> 	     If	the ``publickey'' method is listed more	than once, sshd(8)
> 	     verifies that keys	that have been used successfully are not
> 	     reused for	subsequent authentications.  For example, an
> 	     AuthenticationMethods of ``publickey,publickey'' will require
> 	     successful	authentication using two different public keys.
> 
> 	     This option will yield a fatal error if enabled if	protocol 1 is
> 	     also enabled.  Note that each authentication method listed	should
> 	     also be explicitly	enabled	in the configuration.  The default is
> 	     not to require multiple authentication; successful	completion of
> 	     a single authentication method is sufficient.